### PR TITLE
Move `iree_run_module_test` portability handling to `iree_native_test`

### PR DESCRIPTION
**Pull request by hcindyl**
Define file location with a `{{}}` locator and parse it in `iree_native_test`.

For example, simple.vmfb will be set as 
```
iree_native_test(
...
  ARGS
    "--module_file={{simple.vmfb}}"
...
)
```
And `iree_native_test` handles the portability within the function by check/removing the `{{}}` locator.

### CODE REVIEW

File: build_tools/cmake/iree_native_test.cmake [hunk 1]
```diff
@@ -4,7 +4,31 @@
...
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-include(CMakeParseArguments)
+# iree_parse_input_file_arg
+#
+# A helper function to split the flag and the file path from
+# `TEST_INPUT_FILE_ARGS`. it is for `iree_run_module_test` as `iree-run-module`
+# can have multiple input files while are passed through different flags.
+# The file path is then handled in `iree_native_test` for portability.
+function(iree_parse_input_file_arg OUT_FLAG OUT_FILE INPUT_ARG)
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_native_test.cmake:13]: Oof this is a bit icky. We're taking a string flag and parsing it back out into structured data. How about doing the reverse and passing the flags with placeholders (index of the file arg, perhaps?) and then filling in the appropriate file paths. Either something like

```CMake
ARGS
  "--input={{0}}"
TEST_INPUT_FILES
  "foo.npy"
```

or

```CMake
ARGS
  "--input={{foo.npy}}"
```

(syntax just a strawman). The latter would match Bazel's `$location`

Comment (hcindyl) [reply build_tools/cmake/iree_native_test.cmake]: Done. PTAL.

File: build_tools/cmake/iree_native_test.cmake [hunk 2]
```diff
@@ -89,6 +89,25 @@ function(iree_native_test)
     list(APPEND _RULE_DATA "${_RULE_TEST_INPUT_FILE_ARG}")
```
Comment (pzread) [inline build_tools/cmake/iree_native_test.cmake:89]: No need to do it right now, but we can probably unify _RULE_TEST_INPUT_FILE_ARG and _RULE_ARGS, so all of them use the new `{{}}` syntax.

**Review by pzread — APPROVED** [review]

File: build_tools/cmake/iree_native_test.cmake [hunk 3]
```diff
@@ -89,6 +92,25 @@ function(iree_native_test)
...
+    if(_FILE_ARG)
+      set(_FILE_PATH ${CMAKE_MATCH_1})
+      list(APPEND _RULE_DATA "${_FILE_PATH}")
+      if (ANDROID)
+        cmake_path(GET _FILE_PATH FILENAME _FILE_BASENAME)
+        set(_FILE_PATH "${_ANDROID_ABS_DIR}/${_FILE_BASENAME}")
+      endif()
+      # remove the `{{}}` from `_ARG` and append it to `_TEST_ARGS`.
+      string(REGEX REPLACE "{{.+}}" "" _FILE_FLAG_PREFIX "${_ARG}")
+      list(APPEND _TEST_ARGS "${_FILE_FLAG_PREFIX}${_FILE_PATH}")
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_native_test.cmake:108]: This assumes the file is at the end of the argument, I think. What about

`string(REGEX REPLACE "{{(.+)}}" "\1" _ARG "${_ARG}")`

File: build_tools/cmake/iree_native_test.cmake [hunk 4]
```diff
@@ -89,6 +92,25 @@ function(iree_native_test)
     list(APPEND _RULE_DATA "${_RULE_TEST_INPUT_FILE_ARG}")
   endif()
 
+  # Detect file location with `{{}}` and handle its portability for all entries
+  # in `_RULE_ARGS`.
+  foreach(_ARG ${_RULE_ARGS})
+    string(REGEX MATCH ".+{{(.+)}}" _FILE_ARG "${_ARG}")
+    if(_FILE_ARG)
+      set(_FILE_PATH ${CMAKE_MATCH_1})
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_native_test.cmake:100]: Seems like there could be multiple file paths in a single argument, though it doesn't seem common. I'm not really clear on whether `REGEX MATCH_ALL` is the right way to do that here. I think doing a `MATCH_ALL` without a capture would give you all the matches

```
string(REGEX MATCH_ALL "{{.+}}" _FILES "${_ARG}")
string(REGEX REPLACE "{{(.+)}}" "\\1" _ARG "${_ARG}")

foreach(_FILE IN LISTS _FILES)
  string(REGEX REPLACE "{{(.+)}}" "\\1" _FILE "${_FILE}")
  # adding to data stuff...
endforeach()
```

**Pull request merged**